packages/dapp/src/components/Navbar/NavbarManagement.tsx

```diff
@@ -2,55 +2,58 @@ import MenuIcon from '@mui/icons-material/Menu';
 import { Typography } from '@mui/material';
 import { WalletManagementButtons } from '@transferto/shared/src';
 import { ChainSwitch } from '@transferto/shared/src/atoms/ChainSwitch';
-import { ThemeSwitch } from '@transferto/shared/src/atoms/ThemeSwitch';
-import { useSettings } from '@transferto/shared/src/hooks';
 import { useEffect, useRef } from 'react';
 import { useTranslation } from 'react-i18next';
-import { SubMenuKeys } from '../../const';
+import { shallow } from 'zustand/shallow';
 import { useChainInfos } from '../../providers/ChainInfosProvider';
-import { useMenu } from '../../providers/MenuProvider';
 import { useWallet } from '../../providers/WalletProvider';
+import { useMenuStore, useSettingsStore } from '../../stores';
+import { ThemeSwitch } from '../ThemeSwitch';
 import {
   NavbarDropdownButton,
   NavbarManagement as NavbarManagementContainer,
 } from './Navbar.style';
 
 const NavbarManagement = () => {
-  const anchorRef = useRef<HTMLButtonElement>(null);
-  const settings = useSettings();
-  const menu = useMenu();
+  const mainMenuAnchor = useRef<any>(null);
+
+  const [onWalletDisconnect] = useSettingsStore(
+    (state) => [state.onWalletDisconnect],
+    shallow,
+  );
+
+  const [openMainNavbarMenu, onOpenNavbarMainMenu] = useMenuStore(
+    (state) => [state.openMainNavbarMenu, state.onOpenNavbarMainMenu],
+    shallow,
+  );
+
   const { t: translate } = useTranslation();
   const i18Path = 'navbar.';
   const walletManagement = useWallet();
   const { account } = useWallet();
-
-  !account.isActive ?? settings.onWalletDisconnect();
+  !account.isActive ?? onWalletDisconnect();
 
   // return focus to the button when we transitioned from !open -> open
-  const prevOpen = useRef(menu.openMainNavbarMenu);
+  const prevMainMenu = useRef(openMainNavbarMenu);
   useEffect(() => {
-    if (prevOpen.current === true && menu.openMainNavbarMenu === false) {
-      anchorRef.current!.focus();
+    if (prevMainMenu.current === true && openMainNavbarMenu === false) {
```

Comment:
Probably we can use it like 

`if (prevMainMenu.current && !openMainNavbarMenu) {`